pkg/registry/core/service/storage/rest.go

```diff
@@ -871,6 +865,31 @@ func (al *RESTAllocStuff) initIPFamilyFields(oldService, service *api.Service) e
 	//TODO(thockin): Move this logic to validation?
 	el := make(field.ErrorList, 0)
 
+	// Update-only prep work.
+	if oldService != nil {
+		if getIPFamilyPolicy(service) == api.IPFamilyPolicySingleStack {
+			// As long as ClusterIPs and IPFamilies have not changed, setting
+			// the policy to single-stack is clear intent.
+			if sameClusterIPs(oldService, service) && len(service.Spec.ClusterIPs) > 1 {
+				service.Spec.ClusterIPs = service.Spec.ClusterIPs[0:1]
+			}
+			if sameIPFamilies(oldService, service) && len(service.Spec.IPFamilies) > 1 {
```

Comment:
worth commenting here that ClusterIP[0] is immutable. hence it will always be the correct family.

Reply:
ACK